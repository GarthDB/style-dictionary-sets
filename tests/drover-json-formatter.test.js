const StyleDictionary = require("style-dictionary");
const DroverJsonFormatter = require("../index").DroverJsonFormatter;
const NameKebabTransfom = require("../index").NameKebabTransfom;
const AttributeSetsTransform = require("../index").AttributeSetsTransform;
const helpers = require("./helpers");

const fs = require("fs");
const path = require("path");

StyleDictionary.registerTransform(NameKebabTransfom);
StyleDictionary.registerTransform(AttributeSetsTransform);
StyleDictionary.registerFormat(DroverJsonFormatter);

const generateConfig = (filename) => {
  return {
    source: [`tests/fixtures/${filename}`],
    platforms: {
      DNA: {
        buildPath: helpers.outputDir,
        transforms: [AttributeSetsTransform.name, NameKebabTransfom.name],
        files: [
          {
            destination: filename,
            format: DroverJsonFormatter.name,
            options: {
              showFileHeader: false,
              outputReferences: true,
            },
          },
        ],
      },
    },
  };
};

beforeEach(() => {
  // helpers.clearOutput();
});

afterEach(() => {
  // helpers.clearOutput();
});
test("placeholder", () => {
  expect(true).toBe(true);
})
// test("basic dna json data output", () => {
//   const filename = "basic.json";
//   const sd = StyleDictionary.extend(generateConfig(filename));
//   const output = sd.exportPlatform('DNA');
//   console.log(JSON.stringify(output,'',2));
//   const expected = helpers.fileToJSON(`./tests/expected/${filename}`);
//   expect(output).toMatchObject(expected);
// });
