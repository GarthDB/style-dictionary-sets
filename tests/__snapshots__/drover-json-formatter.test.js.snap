// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`drover format meets basic requirements 1`] = `
Object {
  "KEYS": Object {
    "COLOR_ALIASES": "colorAliases",
    "COLOR_SEMANTICS": "colorSemantics",
    "COLOR_THEMES": "colorThemes",
    "COLOR_TOKENS": "colorTokens",
    "DIMENSIONS": "dimensions",
    "DIMENSION_COMPONENT_LAYOUT_TOKENS": "componentLayoutTokens",
    "DIMENSION_LAYOUT_TOKENS": "layoutTokens",
  },
  "colorThemes": Object {
    "dark": Object {
      "accent-color-100": "rgb(235, 235, 235)",
      "black": "rgb(0, 0, 0)",
      "blackAlias": "rgb(0, 0, 0)",
      "gray-400": "rgb(112, 112, 112)",
      "gray-800": "rgb(235, 235, 235)",
      "neutral-background-color-default": "rgb(112, 112, 112)",
    },
    "darkest": Object {
      "accent-color-100": "rgb(235, 235, 235)",
      "black": "rgb(0, 0, 0)",
      "blackAlias": "rgb(0, 0, 0)",
      "gray-400": "rgb(106, 106, 106)",
      "gray-800": "rgb(235, 235, 235)",
      "neutral-background-color-default": "rgb(106, 106, 106)",
    },
    "light": Object {
      "accent-color-100": "rgb(34, 34, 34)",
      "black": "rgb(0, 0, 0)",
      "blackAlias": "rgb(0, 0, 0)",
      "gray-400": "rgb(177, 177, 177)",
      "gray-800": "rgb(34, 34, 34)",
      "neutral-background-color-default": "rgb(34, 34, 34)",
    },
  },
  "dimensions": Object {
    "another-corner-radius": "2px",
    "base-corner-radius": "2px",
    "component-pill-edge-to-visual-100": "17px",
    "corner-radius-75": "2px",
    "description": "A scale stop of a given size",
    "scale-factor": "1",
  },
  "system": "Spectrum",
  "version": "0.0.0-development",
}
`;

exports[`drover format supports prefix 1`] = `
Object {
  "KEYS": Object {
    "COLOR_ALIASES": "colorAliases",
    "COLOR_SEMANTICS": "colorSemantics",
    "COLOR_THEMES": "colorThemes",
    "COLOR_TOKENS": "colorTokens",
    "DIMENSIONS": "dimensions",
    "DIMENSION_COMPONENT_LAYOUT_TOKENS": "componentLayoutTokens",
    "DIMENSION_LAYOUT_TOKENS": "layoutTokens",
  },
  "colorThemes": Object {
    "dark": Object {
      "aprefix-black": "rgb(0, 0, 0)",
    },
    "darkest": Object {
      "aprefix-black": "rgb(0, 0, 0)",
    },
    "light": Object {
      "aprefix-black": "rgb(0, 0, 0)",
    },
  },
  "dimensions": Object {
    "description": "A scale stop of a given size",
    "scale-factor": "1",
  },
  "system": "Spectrum",
  "version": "0.0.0-development",
}
`;

exports[`drover multi level with repeating color themes 1`] = `
Object {
  "KEYS": Object {
    "COLOR_ALIASES": "colorAliases",
    "COLOR_SEMANTICS": "colorSemantics",
    "COLOR_THEMES": "colorThemes",
    "COLOR_TOKENS": "colorTokens",
    "DIMENSIONS": "dimensions",
    "DIMENSION_COMPONENT_LAYOUT_TOKENS": "componentLayoutTokens",
    "DIMENSION_LAYOUT_TOKENS": "layoutTokens",
  },
  "colorThemes": Object {
    "dark": Object {
      "accent-background-color-default": "rgb(3, 103, 224)",
      "accent-color-500": "rgb(3, 103, 224)",
      "accent-color-600": "rgb(19, 121, 243)",
      "accent-color-900": "rgb(114, 183, 249)",
      "blue-500": "rgb(3, 103, 224)",
      "blue-600": "rgb(19, 121, 243)",
      "blue-900": "rgb(114, 183, 249)",
      "indigo-500": "rgb(85, 91, 231)",
      "indigo-600": "rgb(104, 109, 244)",
      "indigo-900": "rgb(167, 170, 255)",
    },
    "darkest": Object {
      "accent-background-color-default": "rgb(6, 108, 231)",
      "accent-color-500": "rgb(0, 92, 200)",
      "accent-color-600": "rgb(6, 108, 231)",
      "accent-color-900": "rgb(94, 170, 247)",
      "blue-500": "rgb(0, 92, 200)",
      "blue-600": "rgb(6, 108, 231)",
      "blue-900": "rgb(94, 170, 247)",
      "indigo-500": "rgb(73, 78, 216)",
      "indigo-600": "rgb(90, 96, 235)",
      "indigo-900": "rgb(153, 157, 255)",
    },
    "light": Object {
      "accent-background-color-default": "rgb(2, 101, 220)",
      "accent-color-500": "rgb(120, 187, 250)",
      "accent-color-600": "rgb(89, 167, 246)",
      "accent-color-900": "rgb(2, 101, 220)",
      "blue-500": "rgb(120, 187, 250)",
      "blue-600": "rgb(89, 167, 246)",
      "blue-900": "rgb(2, 101, 220)",
      "indigo-500": "rgb(172, 175, 255)",
      "indigo-600": "rgb(149, 153, 255)",
      "indigo-900": "rgb(82, 88, 228)",
    },
  },
  "dimensions": Object {
    "description": "A scale stop of a given size",
    "scale-factor": "1",
  },
  "system": "Spectrum",
  "version": "0.0.0-development",
}
`;
